File: mobile/src/main/java/com/siliconlabs/bledemo/fragment/CharacteristicMappingsFragment.java
package com.siliconlabs.bledemo.fragment;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.siliconlabs.bledemo.mappings.Mapping;
import com.siliconlabs.bledemo.mappings.MappingType;
import com.siliconlabs.bledemo.R;
import com.siliconlabs.bledemo.adapters.MappingAdapter;
import com.siliconlabs.bledemo.utils.SharedPrefUtils;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class CharacteristicMappingsFragment extends Fragment {

    private HashMap<String, Mapping> map;
    private List<Mapping> list;
    private SharedPrefUtils sharedPrefUtils;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        sharedPrefUtils = new SharedPrefUtils(getActivity());
        map = sharedPrefUtils.getCharacteristicNamesMap();
        list = new ArrayList<>(map.values());

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_characteristic_mappings, container, false);

        RecyclerView recyclerView = view.findViewById(R.id.recycler_view);
        MappingAdapter adapter = new MappingAdapter(list, getActivity(), MappingType.CHARACTERISTIC);
        recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
        recyclerView.setAdapter(adapter);

        // Inflate the layout for this fragment
        return view;
    }

    @Override
    public void onPause() {
        super.onPause();

        map.clear();

        for (Mapping mapping : list) {
            map.put(mapping.getUuid(), mapping);
        }

        sharedPrefUtils.saveCharacteristicNamesMap(map);
    }
}

File: mobile/src/main/java/com/siliconlabs/bledemo/Browser/Fragments/CharacteristicMappingsFragment.kt
package com.siliconlabs.bledemo.Browser.Fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import com.siliconlabs.bledemo.Browser.Adapters.MappingAdapter
import com.siliconlabs.bledemo.Browser.Models.Mapping
import com.siliconlabs.bledemo.Browser.Models.MappingType
import com.siliconlabs.bledemo.R
import com.siliconlabs.bledemo.Utils.SharedPrefUtils
import kotlinx.android.synthetic.main.fragment_characteristic_mappings.*
import java.util.*
import kotlin.collections.ArrayList

class CharacteristicMappingsFragment : Fragment() {
    private lateinit var map: HashMap<String, Mapping>
    private lateinit var list: ArrayList<Mapping>
    private lateinit var sharedPrefUtils: SharedPrefUtils

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        sharedPrefUtils = SharedPrefUtils(requireContext())
        map = sharedPrefUtils.characteristicNamesMap
        list = ArrayList(map.values)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_characteristic_mappings, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        rv_characteristics.layoutManager = LinearLayoutManager(activity)
        rv_characteristics.adapter = MappingAdapter(list, requireContext(), MappingType.CHARACTERISTIC)
    }

    override fun onPause() {
        super.onPause()
        map.clear()
        for (mapping in list) map[mapping.uuid] = mapping
        sharedPrefUtils.saveCharacteristicNamesMap(map)
    }
}